### PR TITLE
No more QueryOutputType

### PR DESCRIPTION
**Pull request by danxmoran**
### Description

This cleanup brought to you by cool ideas from NE Scala that I want to try, but not until we clean house.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-transfer-model/src/main/scala/org/broadinstitute/clio/util/json/JsonSchema.scala [hunk 1]
```diff
@@ -3,84 +3,53 @@ package org.broadinstitute.clio.util.json
...
+  * A JSON schema for a key / metadata pair in Clio.
   *
-  * JsonSchema uses the same shapeless pattern as the
-  * [[org.broadinstitute.clio.util.generic.FieldMapper]]
-  *
-  * @tparam A is a case class suitable for Circe such as a QueryOutput
+  * @tparam CI The ClioIndex of the key / metadata pair.
   */
-trait JsonSchema[A] {
+class JsonSchema[CI <: ClioIndex](index: CI) {
```
Comment (danxmoran) [inline clio-transfer-model/src/main/scala/org/broadinstitute/clio/util/json/JsonSchema.scala]: While I'm ripping stuff out... nobody actually uses the `get-schema` functionality. Can we get rid of it, too?

Comment (pshapiro4broad) [reply clio-transfer-model/src/main/scala/org/broadinstitute/clio/util/json/JsonSchema.scala]: Seems ok to me

File: clio-transfer-model/src/main/scala/org/broadinstitute/clio/util/json/JsonSchema.scala [hunk 2]
```diff
@@ -98,89 +67,44 @@ object JsonSchema {
...
+      case t if t =:= typeOf[Option[String]]         => aString
+      case t if t =:= typeOf[Option[Float]]          => aFloat
+      case t if t =:= typeOf[Option[Double]]         => aDouble
+      case t if t <:< typeOf[Option[EnumEntry]]      => aString
+      case t if t =:= typeOf[Option[OffsetDateTime]] => aTime
+      case t if t =:= typeOf[Option[Symbol]]         => aString
+      case t if t =:= typeOf[Option[UUID]]           => aString
+      case t if t =:= typeOf[Option[URI]]            => aUri
+      case t if t <:< typeOf[Option[Seq[String]]]    => aStringArray
+      case t if t <:< typeOf[Option[Seq[URI]]]       => aUriArray
```
Comment (pshapiro4broad) [inline clio-transfer-model/src/main/scala/org/broadinstitute/clio/util/json/JsonSchema.scala]: 👍

**Review by pshapiro4broad — APPROVED** [review]

**Comment by pshapiro4broad** [comment]
It looks like this PR reduces coverage of `CaseClassMapper`, maybe this is because the remaining uses of it are in clio-client which is mainly covered by integration tests, which aren't included in the coverage results.

File: clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala [hunk 3]
```diff
@@ -91,6 +92,6 @@ class ClioWebClientSpec
   }
 
   it should "retry requests that fail with connection errors" in {
-    client.getSchema(WgsUbamIndex).map(_ should be(WgsUbamIndex.jsonSchema))
+    client.query(index)(ModelMockQueryInput(), includeDeleted = false).map(_ => succeed)
```
Comment (charlesdyan) [inline clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala]: How does this test retry on connection error? I think I'm missing something

**Review by charlesdyan — APPROVED** [review]

File: clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala (hunk 3 above)
Comment (danxmoran) [reply clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala]: The mock server is set up to time out this specific route N-1 times... I'll add a comment.

**Pull request merged**